pkg/kubelet/volume/gce_pd/gce_pd.go

```diff
@@ -263,7 +263,14 @@ func (pd *gcePersistentDisk) TearDown() error {
 	// If refCount is 1, then all bind mounts have been removed, and the
 	// remaining reference is the global mount. It is safe to detach.
 	if refCount == 1 {
-		if err := pd.manager.DetachDisk(pd, devicePath); err != nil {
+		// pd.pdName is not initially set for volume-cleaners, so set it here.
+		var err error
+		globalMountPath := path.Join(pd.plugin.host.GetPluginDir(gcePersistentDiskPluginName), "mounts")
+		pd.pdName, err = getPDName(globalMountPath, pd.mounter, devicePath)
```

Comment:
This will cause a second iteration of the whole mount list, which is potentially very expensive.  The main reason to return the device path up above was to use it to unmount without iterating the mount list again.